install.sh

```diff
@@ -20,67 +20,89 @@ file_issue_prompt() {
+  res=$(curl -fsSL https://api.github.com/repos/flipt-io/flipt/releases/latest | grep tag_name | cut -d '"' -f 4)
   echo "$res"
 }
 
 copy() {
-  if [[ ":$PATH:" == *":$HOME/.local/bin:"* ]]; then
-      if [ ! -d "$HOME/.local/bin" ]; then
-        mkdir -p "$HOME/.local/bin"
-      fi
+  # if [ ":$PATH:" = *":$HOME/.local/bin:"* ]; then
+  #     [ ! -d "$HOME/.local/bin" ] && mkdir -p "$HOME/.local/bin"
+  #     mv /tmp/flipt "$HOME/.local/bin/flipt"
+  # else
+  #     # Try without sudo first, run with sudo only if mv failed without it.
+  #     mv /tmp/flipt /usr/local/bin/flipt || (
+  #       echo "Cannot write to installation target directory as current user, writing as root."
+  #       sudo mv /tmp/flipt /usr/local/bin/flipt
+  #     )
+  # fi
+
+  case ":PATH:" in 
+    *":HOME/.local/bin:"*) 
+      [ ! -d "$HOME/.local/bin" ] && mkdir -p "$HOME/.local/bin"
       mv /tmp/flipt "$HOME/.local/bin/flipt"
-  else
-      # Try without sudo first, run with sudo only if mv failed without it.
+      ;;
+    *)
       mv /tmp/flipt /usr/local/bin/flipt || (
-        echo "Cannot write to installation target directory as current user, writing as root."
-        sudo mv /tmp/flipt /usr/local/bin/flipt
-      )
-  fi
+      echo "Cannot write to installation target directory as current user, writing as root."
+      sudo mv /tmp/flipt /usr/local/bin/flipt
+    )
+    ;;
+  esac 
 }
 
 # This function decides what version will be installed based on the following priority:
 # 1. Environment variable `VERSION` is set.
 # 2. Latest available on GitHub
-function get_version() {
-  if [[ -z "$VERSION" ]]; then
-      if [[ -n "$1" ]]; then
+get_version() {
+  if [ -z "$VERSION" ]; then
+      if [ -n "$1" ]; then
           VERSION="$1"
       else
           VERSION=$(get_latest_version)
       fi
   fi
+  
   # ensure version starts with v
-  if [[ "$VERSION" != v* ]]; then
-    VERSION="v$VERSION"
-  fi
+  case "$VERSION" in
+    v*) : ;;
+    *) VERSION="v$VERSION" ;;
+  esac
   echo "$VERSION"
 }
 
-function install() {
-  local version=$(get_version $1);
+install() {
+  version=$(get_version "$1");
   echo "Installing version $version"
-  if [[ "$OSTYPE" == "linux"* ]]; then
-      ARCH=$(uname -m);
-      OS="linux";
-      if [[ "$ARCH" != "x86_64" && "$ARCH" != "aarch64" ]]; then
+
+  OSCHECK=$(uname | tr '[:upper:]' '[:lower:]')
+  
+  case "$OSCHECK" in 
+    linux*)
+      ARCH=$(uname -m)
+      OS="linux"
+      case "$ARCH" in 
+        x86_64|aarch64) : ;; 
+        *)
           echo "flipt is only available for linux x86_64/arm64 architecture"
           file_issue_prompt
+          ;; 
+      esac 
+      [ "$ARCH" = "aarch64" ] && ARCH="arm64"
+      ;;
+    darwin*) 
+        ARCH=$(uname -m)
+        OS="darwin"
+        if [ "$ARCH" != "arm64" ]; then
+        echo "flipt is only available for mac arm64 architecture"
```

Comment:
I know you just copied this as it was, but while you're here, mind updating this as we have arm build for Darwin now 🙏

Reply:
Yup, I have done it in the latest commit ✨ for Apple Silicon ARM64, also tested on my machine.

Reply:
Thank you @PythonHacker24 🙌 Legendary!

Reply:
Signed off the commits as per the link suggested and thanks for the compliment! ✨

Reply:
This is awesome, thanks again.